### PR TITLE
Implement crypto libs and remove neofs-crypto usage

### LINKED ISSUE #317: Drop neofs-crypto
I think it will be nice to drop neofs-crypto before we go to neofs-sdk-go. RFC6979 verification we can use directly from neo-go, but other ECDSA check should be moved there. 

Related to https://github.com/nspcc-dev/neofs-node/issues/491

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Closes #317.

### CODE REVIEW

File: crypto/proto/sign.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	StableMarshal([]byte) error
+
+	// StableSize returns size of protobuf data.f
+	//
+	// Size must not be negative.
+	StableSize() int
+}
+
+// BufferAllocator is a function which allocates
+// byte slices of particular length,
```
Comment (cthulhu-rider) [inline crypto/proto/sign.go:23]:
```suggestion
// byte slices of particular length.
```

File: crypto/rfc6979/rfc6979.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	"math/big"
+	"sync"
+
+	neofscrypto "github.com/nspcc-dev/neofs-api-go/crypto"
+	"github.com/nspcc-dev/rfc6979"
+)
+
+type rfc6979Signer ecdsa.PrivateKey
+
+// Signer provides neofscrypto.Signer interface of the ecdsa,PrivateKey.
```
Comment (cthulhu-rider) [inline crypto/rfc6979/rfc6979.go:17]:
```suggestion
// Signer provides neofscrypto.Signer interface of the ecdsa.PrivateKey.
```

File: pkg/client/object.go [hunk 3]
```diff
@@ -826,15 +827,18 @@ func (c *clientImpl) GetObjectHeader(ctx context.Context, p *ObjectHeaderParams,
...
+			return nil, err
+		}
+
+		var vPrm apicrypto.VerifyPrm
+
+		vPrm.SetProtoMarshaler(signature.StableMarshalerCrypto(p.addr.ObjectID().ToV2()))
+		vPrm.SetSignature(idSig.GetSign())
+
+		if !apicrypto.Verify(key, vPrm) {
+			return nil, fmt.Errorf("incorrect object header signature")
```
Comment (cthulhu-rider) [inline pkg/client/object.go:841]:
```suggestion
			return nil, errors.New("incorrect object header signature")
```

File: crypto/algo/keys.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+// UnmarshalKey unmarshals neofscrypto.PublicKey from a binary representation
+// according to SignatureAlgorithm.
+//
+// If algo is not supported, an error returns.
+func UnmarshalKey(algo SignatureAlgorithm, binKey []byte) (neofscrypto.PublicKey, error) {
+	c, ok := v.m[algo]
+	if !ok {
+		return nil, errors.New("unsupported signature algorithm")
```
Comment (carpawell) [inline crypto/algo/keys.go:37]: would it be better to make it var error or format error with passed unsupported algo?

File: crypto/ecdsa/ecdsa.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+)
+
+type ecdsaSigner ecdsa.PrivateKey
+
+var (
+	onceCurve sync.Once
+	curve     elliptic.Curve
+)
+
+// Signer provides neofscrypto.Signer interface of the ecdsa,PrivateKey.
```
Comment (carpawell) [inline crypto/ecdsa/ecdsa.go:22]:
```suggestion
// Signer provides neofscrypto.Signer interface of the ecdsa.PrivateKey.
```

File: crypto/ecdsa/ecdsa.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+	"math/big"
+	"sync"
+
+	neofscrypto "github.com/nspcc-dev/neofs-api-go/crypto"
+)
+
+type ecdsaSigner ecdsa.PrivateKey
+
+var (
+	onceCurve sync.Once
```
Comment (carpawell) [inline crypto/ecdsa/ecdsa.go:18]: can't it be done with `init()` func here and beyong?

**Review by carpawell — CHANGES_REQUESTED** [review]

File: crypto/algo/keys.go (hunk 4 above)
Comment (cthulhu-rider) [reply crypto/algo/keys.go:37]: No error is expected if done correctly. Trying to fix it at runtime is pointless. At the same time, at first glance, the panic did not seem justified to me, at least from the server side. I suggest to leave it as it is until we decide to panic.

File: crypto/ecdsa/ecdsa.go (hunk 6 above)
Comment (cthulhu-rider) [reply crypto/ecdsa/ecdsa.go:18]: It can be, but not all public functionality requires this initialization, so in special cases it will be redundant.

**Review by alexvanin — APPROVED** [review]
Naming and package organization will be discussed later before v1.29.0 release, I guess.

**Review by carpawell — APPROVED** [review]

**Pull request merged**